### PR TITLE
wps_hide_login module CVE-2021-24917

### PR DESCRIPTION
**Pull request by h00die**
This module is basically a glorified `curl` command. it should take 10min to test .

## Verification

List the steps needed to make sure this thing works

- [x] Install the vulnerable plugin and set a new login page
- [x] Start msfconsole
- [x] Do: `use auxiliary/scanner/http/wp_wps_hide_login_revealer`
- [x] Do: `set rhosts [ip]`
- [x] Do: `run`
- [x] You should find the hidden login page
- [x] **Document** is good

### CODE REVIEW

**Review by gwillcox-r7 — CHANGES_REQUESTED** [review]
Overall looks good, some very minor suggestions for improvement. Nice work :)

**Comment by gwillcox-r7** [comment]
> you poked the bull, so enjoy the horns :) you had a good comment about the 'vulnerable version detected'... OF WHAT. So i went ahead and updated EVERY module that had that phrase in it.

XD I suppose I did ask for that XD Will review this now

File: modules/auxiliary/scanner/http/wp_learnpress_sqli.rb [hunk 1]
```diff
@@ -63,7 +63,7 @@ def run_host(ip)
       vprint_error('Learnpress version not vulnerable')
       return
     end
-    print_good('Vulnerable version detected')
+    print_good('Vulnerable version of Learnpress detected')
```
Comment (gwillcox-r7) [inline modules/auxiliary/scanner/http/wp_learnpress_sqli.rb:66]:
```suggestion
    print_good('Vulnerable version of LearnPress detected')
```

File: modules/auxiliary/scanner/http/wp_email_sub_news_sqli.rb [hunk 2]
```diff
@@ -50,7 +55,7 @@ def run_host(ip)
     unless [Msf::Exploit::CheckCode::Vulnerable, Msf::Exploit::CheckCode::Appears, Msf::Exploit::CheckCode::Detected].include?(checkcode)
       fail_with Failure::NotVulnerable, 'Email subscribers and newsletter version not vulnerable'
```
Comment (gwillcox-r7) [inline modules/auxiliary/scanner/http/wp_email_sub_news_sqli.rb:56]:
```suggestion
      fail_with Failure::NotVulnerable, 'Email Subscribers and Newsletter version not vulnerable'
```

File: modules/auxiliary/scanner/http/wp_email_sub_news_sqli.rb [hunk 3]
```diff
@@ -50,7 +55,7 @@ def run_host(ip)
...
     end
-    print_good('Vulnerable version detected')
+    print_good('Vulnerable version of Email subscribers and newsletter detected')
```
Comment (gwillcox-r7) [inline modules/auxiliary/scanner/http/wp_email_sub_news_sqli.rb:58]:
```suggestion
    print_good('Vulnerable version of Email Subscribers and Newsletter detected')
```

File: modules/auxiliary/scanner/http/wp_total_upkeep_downloader.rb [hunk 4]
```diff
@@ -44,9 +49,9 @@ def run_host(ip)
 
     checkcode = check_plugin_version_from_readme('boldgrid-backup', '1.14.10')
     unless [Msf::Exploit::CheckCode::Vulnerable, Msf::Exploit::CheckCode::Appears, Msf::Exploit::CheckCode::Detected].include?(checkcode)
-      fail_with Failure::NotVulnerable, "#{ip} - A vulnerable version of the 'Boldgrid Backup' was not found"
+      fail_with Failure::NotVulnerable, "#{ip} - A vulnerable version of 'Boldgrid Backup' was not found"
     end
-    print_good("#{ip} - Vulnerable version detected")
+    print_good("#{ip} - Vulnerable version of 'Boldgrid Backup' detected")
```
Comment (gwillcox-r7) [inline modules/auxiliary/scanner/http/wp_total_upkeep_downloader.rb:54]:
```suggestion
    print_good("#{ip} - Vulnerable version of Boldgrid Backup detected")
```

File: modules/auxiliary/scanner/http/wp_total_upkeep_downloader.rb [hunk 5]
```diff
@@ -44,9 +49,9 @@ def run_host(ip)
 
     checkcode = check_plugin_version_from_readme('boldgrid-backup', '1.14.10')
     unless [Msf::Exploit::CheckCode::Vulnerable, Msf::Exploit::CheckCode::Appears, Msf::Exploit::CheckCode::Detected].include?(checkcode)
-      fail_with Failure::NotVulnerable, "#{ip} - A vulnerable version of the 'Boldgrid Backup' was not found"
+      fail_with Failure::NotVulnerable, "#{ip} - A vulnerable version of 'Boldgrid Backup' was not found"
```
Comment (gwillcox-r7) [inline modules/auxiliary/scanner/http/wp_total_upkeep_downloader.rb:52]:
```suggestion
      fail_with Failure::NotVulnerable, "#{ip} - A vulnerable version of Boldgrid Backup was not found"
```

File: documentation/modules/auxiliary/scanner/http/wp_total_upkeep_downloader.md [hunk 6]
```diff
@@ -36,7 +36,7 @@ resource (total_upkeep.rb)> use auxiliary/scanner/http/wp_total_upkeep_downloade
 resource (total_upkeep.rb)> set rhosts 1.1.1.1
 rhosts => 1.1.1.1
 resource (total_upkeep.rb)> run
-[+] 1.1.1.1 - Vulnerable version detected
+[+] 1.1.1.1 - Vulnerable version of 'Boldgrid Backup' detected
```
Comment (gwillcox-r7) [inline documentation/modules/auxiliary/scanner/http/wp_total_upkeep_downloader.md:39]:
```suggestion
[+] 1.1.1.1 - Vulnerable version of Boldgrid Backup detected
```

**Review by gwillcox-r7 — APPROVED** [review]
Ignore these other comments @h00die they are purely for my tracking purposes to ensure I fix them before this gets merged in. I will go ahead and make these changes since you took the time to try and fix what I pointed out.

**Comment by h00die** [comment]
appreciated, but thats not how teamwork works. I got it :)

File: documentation/modules/auxiliary/scanner/http/wp_learnpress_sqli.md [hunk 7]
```diff
@@ -60,7 +60,7 @@ count => 3
 resource (learnpress.rb)> run
 [*] Checking /wp-content/plugins/learnpress/readme.txt
 [*] Found version 3.2.6.7 in the plugin
-[+] Vulnerable version detected
+[+] Vulnerable version of Learnpress detected
```
[the remark this answers is not in the document]
Comment (gwillcox-r7) [reply documentation/modules/auxiliary/scanner/http/wp_learnpress_sqli.md]: This was corrected but you removed the `n` here. I'll fixed this before landing.

**Comment by gwillcox-r7** [comment]
For reference to download and install the plugin:

1. Go to https://downloads.wordpress.org/plugin/wps-hide-login.1.9.zip
2. Log in to WordPress
3. Go to `Plugins`->`Add New` and click `Upload Plugin` and select the downloaded ZIP file.
4. Click `Install Now`
5. Click `Activate Plugin` on the install page confirming the install succeeded.
6. All good, start testing!

Will add these in to readme just so people know how to find the file since it may not be obvious to those not familiar with Wordpress how to obtain older versions of a plugin.

**Comment by gwillcox-r7** [comment]
Also to edit plugin settings, go to `/wp-admin/plugins.php`, find the `WPS Hide Login` plugin, click `Settings` and then change `Login URL`.

Alternatively just go to `/wp-admin/options-general.php#whl_settings` and change it there.

**Comment by gwillcox-r7** [comment]
Testing completed, looks like it works as expected, will add in some directions for additional info on setting up a test environment and fix the minor typo as part of the PR land but this looks good to go. Landing shortly.

```
msf6 > use auxiliary/scanner/http/wp_wps_hide_login_revealer
msf6 auxiliary(scanner/http/wp_wps_hide_login_revealer) > set RHOSTS 127.0.0.1
RHOSTS => 127.0.0.1
msf6 auxiliary(scanner/http/wp_wps_hide_login_revealer) > show options

Module options (auxiliary/scanner/http/wp_wps_hide_login_revealer):

   Name       Current Setting  Required  Description
   ----       ---------------  --------  -----------
   Proxies                     no        A proxy chain of format type:host:port[,type:hos
                                         t:port][...]
   RHOSTS     127.0.0.1        yes       The target host(s), see https://github.com/rapid
                                         7/metasploit-framework/wiki/Using-Metasploit
   RPORT      80               yes       The target port (TCP)
   SSL        false            no        Negotiate SSL/TLS for outgoing connections
   TARGETURI  /                yes       The base path to the wordpress application
   THREADS    1                yes       The number of concurrent threads (max one per ho
                                         st)
   VHOST                       no        HTTP server virtual host

msf6 auxiliary(scanner/http/wp_wps_hide_login_revealer) > set RPORT 8000
RPORT => 8000
msf6 auxiliary(scanner/http/wp_wps_hide_login_revealer) > run

[+] 127.0.0.1 - Vulnerable version of wps_hide_login detected
[*] 127.0.0.1 - Determining login page
[+] Login page: http://127.0.0.1:8000/?whatasecretloginpage&redirect_to=%2Fwp-admin%2FD7veG&reauth=1
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/http/wp_wps_hide_login_revealer) > 
```

**Comment by gwillcox-r7** [comment]
For reference https://github.com/rapid7/metasploit-framework/commit/ace37bd6780a451981b4632eabe9e2d1bf8eee01 was ninja added to fix up typo and to add a bit more documentation around how to install WordPress and download and configure the plugin just in case it helps anyone.

**Comment by gwillcox-r7** [comment]
# Release Notes
A new PR for CVE-2021-24917 was added, which is an information disclosure bug in WPS Hide Login WordPress plugin before 1.9.1. This vulnerability allows unauthenticated users to get the secret login page by setting a random referer string and making a request to `/wp-admin/options.php`. Additionally, several WordPress modules were updated to more descriptively report which plugin they found as being vulnerable on a given target.

**Pull request merged**